teps/0154-concise-remote-resolver-syntax.md

```diff
@@ -0,0 +1,308 @@
+
+```yaml
+taskRef:
+  resolver: "cluster"
+  name: "cluster://task/default/git-clone" 
+```
+
+### Stage 2
+
+In this stage, we enable the Tekton controller to be able to identify the resolvers on the clusters that have subscribed to the scheme of the incoming [resolver reference](#resolver-reference).
+
+#### Subscribing to `scheme`s
+
+Each remote resolver will subscribe to `scheme`s. e.g. the git resolver might subscribe to `scheme`s like `git+https`, `https` or a bundle resolver might subscribe to `scheme`s like `oci`.
+This mapping of `scheme` to `resolver` could be easily maintained in a `configMap` for the cluster. The cluster operators will be required to know the remote resolvers that are running on the cluster and the schemes they are subscribed to.  
+The configMap must list out all the schemes supported by all the resolvers.
+ 
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: remote-resolution-scheme-subscription
+  namespace: tekton-pipelines
+data:
+  #<scheme>: "<resolvers subscribed to the scheme; comma separated>"
+  https: "git,http"
+  git+https: "git"
+  git: "git"
+  http: "http"
+  oci: "bundle"
+  hub: "hub"
+```
+
+#### Prioritizing the order of resolvers
+
+Occasionally, for certain common `scheme`s like `https`, multiple resolvers could be subscribed to it. In that case, the Tekton controller would send out requests based on the priority to the list of the subscribed resolvers until one of them succedes in resolving the request. The cluster operators will maintain the priority of the resolvers based on the order of the resolvers subscribed to the scheme. In the comma-separated value, the left side has the higher priority. The advantage here is that depending on the `scheme`, the priority can be different and is under the control of the cluster operators.
+
+####  Creating the resolution request
+The when the Tekton controller (`PipelineRun` or `TaskRun`) creates a resolution request, it will match the `scheme` to the prioritized resolvers and create a new resolution request for each resolver until one succeedes or they all fail. If a `scheme` is not supported by any of the resolvers on the cluster then the Tekton controller will directly fail the `TaskRun/PipelineRun`. If all of the resolvers subscribed to the scheme fail to resolve it then the Tekton controller will also fail the `TaskRun/PipelineRun`.
+
+## Design Details
+The feature will be gated behind a feature flag: `enable-concise-resolver-syntax`. The feature will be propagated through the lifecycle of `alpha --> beta --> stable` like we do with other features.
+
+### Updating the `name` field
+* Currently the value of the `name` field must be a sub-domain name. We will loosen this restricitons to also allow for a url-like name. 
+* The string field `name` will also be added to the `ResolutionRequest CRD` which will contain the value (after variable replacement) passed without any validation from `taskRef, pipelineRef and ref`.
```

Comment:
I thought this section was talking about the existing name field in the CRD. 😅 

Since `name` is generic, can you link out to the specific types you want to modify here?

Reply:
Yes, it's a bit confusing. I will clarify it.

We have a `name` in the `TaskRef`, `PipelineRef` and `Ref` (StepAction's):
```
taskRef:
  - name: blah (currently to resolver locally applied tasks)
```
I think this confusion will go away if we add a new field `url` like you suggested above instead of repurposing the field `name`.